api/src/main/java/org/openmrs/api/db/hibernate/HibernateUserDAO.java

```diff
@@ -652,6 +667,26 @@ private Query createUserSearchQuery(String name, List<Role> roles, boolean inclu
 		
 		return query;
 	}
+
+	/**
+	 * @see org.openmrs.api.db.UserDAO#createActivationKey(org.openmrs.User)
+	 */
+	@Override
+	public void createActivationKey(User user) {
+		log.info("Setting activationKey for " + user.getUsername());
+			int validTime = 10*60*1000; //token should be valid for 10 minutes
+			String token = RandomStringUtils.randomAlphanumeric(20);
+			long time = System.currentTimeMillis() + validTime;
+			String hashedKey = Security.encodeString(token);
+			String activationKey = hashedKey+":"+time;
+			
+			LoginCredential credentials = getLoginCredential(user);
+			credentials.setActivationKey(activationKey);			
+			sessionFactory.getCurrentSession().merge(credentials);
+			//Send Email with unhashed  activation key and Date:
```

Comment:
Even sending of email should be in the service layer so remove this comment, the DAO should only be used to do database operations only.